django/forms/models.py

```diff
@@ -52,6 +52,10 @@ def construct_instance(form, instance, fields=None, exclude=None):
             continue
         if exclude and f.name in exclude:
             continue
+        # Leave defaults for fields that aren't in POST data, except for
+        # checkbox inputs because they don't appear in POST data if not checked.
+        if f.has_default() and f.name not in form.data and not isinstance(form[f.name].field.widget, CheckboxInput):
```

Comment:
I'm not too happy with this handling of checkboxes, but this at least starts the discussion...